### PR TITLE
Tree shake (dead code elimination)

### PR DESCRIPTION
**Pull request by chrisnojima**
This PR applys some tree shaking to find unused code

The process is
1. `yarn build-treeshake`
1. Edit desktop/dist/*.bundle.js and look for  `unused harmony` comments
1. Search and remove 
1. Run flow / lint check (treeshake doesn't understand native so if its some export thats only used by a .native.js file it'll appear as unused)

### CODE REVIEW

File: shared/actions/json/login.json [hunk 1]
```diff
@@ -11,16 +11,11 @@
     "relogin": {
       "usernameOrEmail": "string",
       "passphrase": "HiddenString",
-      "canError": {
```
Comment (chrisnojima) [inline shared/actions/json/login.json]: we never emit errors for this

File: shared/actions/json/signup.json [hunk 2]
```diff
@@ -44,11 +44,8 @@
     "showPaperKey": {
       "paperkey": "HiddenString"
     },
-    "showSuccess": {},
-    "signup": {
```
Comment (chrisnojima) [inline shared/actions/json/signup.json]: we only ever emit errors for this so just make it a regular action

File: shared/desktop/webpack.config.babel.js [hunk 3]
```diff
@@ -22,6 +22,7 @@ const flags = {
   isHot: getenv.boolish('HOT', false),
   isShowingDashboard: !getenv.boolish('NO_SERVER', !isDev),
   isVisDiff: getenv.boolish('VISDIFF', false),
+  isTreeShake: getenv.boolish('TREESHAKE', false),
```
Comment (chrisnojima) [inline shared/desktop/webpack.config.babel.js]: make it aware of a tree shake build

**Comment by chrisnojima** [comment]
- [x] mobile works
- [x] desktop works

**Comment by chrisnojima** [comment]
@keybase/react-hackers this pr removes some unused code. If there's something that looks like it should still be there, let me know cause everything compiles and works so the calling function is likely not there anymore and should go back in

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: shared/actions/teams/index.js [hunk 4]
```diff
@@ -387,21 +387,6 @@ const _getDetails = function*(action: Types.GetDetails): Saga.SagaGenerator<any,
   }
 }
 
-const _changeOpenTeamSetting = function*({
```
Comment (MarcoPolo) [inline shared/actions/teams/index.js]: We def. use this, maybe it's because we're using a constant below that it doesn't realize that?

Comment (chrisnojima) [reply shared/actions/teams/index.js]: the constant doesn't exist, it was added back in by cjb 2 days ago (before this pr)

Comment (chrisnojima) [reply shared/actions/teams/index.js]: oh wait no, we never actually dispatch that action, so the below is removed, and now this is removed

**Pull request merged**